workspaces-tests/tests/upgrade.rs

```diff
@@ -0,0 +1,206 @@
+#![cfg(not(windows))]
+
+use near_contract_tools::approval::native_transaction_action::PromiseAction;
+use near_sdk::{serde_json::json, Gas};
+use workspaces::{network::Sandbox, prelude::*, Account, Contract, Worker};
+
+const WASM: &[u8] = include_bytes!("../../target/wasm32-unknown-unknown/release/upgrade_old.wasm");
+
+const NEW_WASM: &[u8] =
+    include_bytes!("../../target/wasm32-unknown-unknown/release/upgrade_new.wasm");
+
+const BAD_WASM: &[u8] =
+    include_bytes!("../../target/wasm32-unknown-unknown/release/upgrade_bad.wasm");
+
+const RANDOM_WASM: &[u8] =
+    include_bytes!("../../target/wasm32-unknown-unknown/release/counter_multisig.wasm");
+
+struct Setup {
+    pub worker: Worker<Sandbox>,
+    pub contract: Contract,
+    pub accounts: Vec<Account>,
+}
+
+/// Setup for individual tests
+async fn setup(num_accounts: usize, wasm: &[u8]) -> Setup {
+    let worker = workspaces::sandbox().await.unwrap();
+
+    // Initialize contract
+    // let contract = worker.dev_deploy(&wasm.to_vec()).await.unwrap();
+    // contract.call(&worker, "new").transact().await.unwrap();
+
+    // Initialize user accounts
+    let mut accounts = vec![];
+    for _ in 0..(num_accounts + 1) {
+        accounts.push(worker.dev_create_account().await.unwrap());
+    }
+
+    let alice = &accounts[0].clone();
+
+    let contract = alice
+        .deploy(&worker, &wasm.to_vec())
+        .await
+        .unwrap()
+        .unwrap();
+    contract.call(&worker, "new").transact().await.unwrap();
+
+    Setup {
+        worker,
+        contract,
+        accounts,
+    }
+}
+#[tokio::test]
+async fn upgrade() {
+    // Deploy old contract ** LIKE HERE
+    // Initialize old contract
+    let Setup {
+        worker,
+        contract,
+        accounts,
+    } = setup(3, WASM).await;
+
+    let alice = &accounts[0];
+    let bob = &accounts[1];
+    let charlie = &accounts[2];
+
+    alice
+        .call(&worker, contract.id(), "increment_foo")
+        .args_json(json!({}))
+        .unwrap()
+        .transact()
+        .await
+        .unwrap();
+
+    let val = alice
+        .call(&worker, contract.id(), "get_foo")
```

Comment:
This can be a `view` call, as can the `get_bar` calls below

Reply:
created this issue to clean up after: https://github.com/NEARFoundation/near-contract-tools/issues/67